### PR TITLE
Create-Task-Extract-Data

### PR DESCRIPTION
**Pull request by AleBarboni**

### CODE REVIEW

File: src/components/modal-content/CreateTask.jsx [hunk 1]
```diff
@@ -28,12 +28,57 @@ const OPTIONS = [
...
+			Jun: "06",
+			Jul: "07",
+			Aug: "08",
+			Sep: "09",
+			Oct: "10",
+			Nov: "11",
+			Dec: "12",
+		};
+		return [mnths[date[2]], date[1], date[3]].join("/");
+	}
```
Comment (cezarmocanu) [inline src/components/modal-content/CreateTask.jsx:51]: This function is not needed as we can transform the selected date to js date using `date.toDate()`

File: src/components/modal-content/CreateTask.jsx [hunk 2]
```diff
@@ -28,12 +28,57 @@ const OPTIONS = [
...
+
+	const [taskName, setTaskName] = useState("");
+	const [priority, setPriority] = useState("");
+	const [dueDate, setDueDate] = useState("");
+	const [description, setDescription] = useState("");
```
Comment (cezarmocanu) [inline src/components/modal-content/CreateTask.jsx:56]: Having multiple fields with separate states is very work intensive and leads to multiple event handlers. It is enough to have a single formState object such as
```js
{
  taskName: string,
  priority: string,
  dueDate: Date,
  description: string,
}

```

File: src/components/modal-content/CreateTask.jsx [hunk 3]
```diff
@@ -28,12 +28,57 @@ const OPTIONS = [
...
 	const dispatch = useDispatch();
```
Comment (cezarmocanu) [inline src/components/modal-content/CreateTask.jsx:57]: Dispatch is not needed as it does not call any redux actions

File: src/components/modal-content/CreateTask.jsx [hunk 4]
```diff
@@ -28,12 +28,57 @@ const OPTIONS = [
 ];
 
 function CreateTask() {
-	const [select, setSelect] = useState("");
+	const today = dayjs();
+	const tomorrow = dayjs().add(1, "day");
```
Comment (cezarmocanu) [inline src/components/modal-content/CreateTask.jsx:32]: add useMemo as they should not be recomputed every render

File: src/components/modal-content/CreateTask.jsx [hunk 5]
```diff
@@ -28,12 +28,57 @@ const OPTIONS = [
...
 
-	const handleChangeSelect = (event) => {
-		setSelect(event.target.value);
+	const handleChangeTaskName = (event) => {
+		setTaskName(event.target.value);
+	};
+	const handleChangePriority = (event) => {
+		setPriority(event.target.value);
+	};
+	const handleChangeDueDate = (selectDate) => {
```
Comment (cezarmocanu) [inline src/components/modal-content/CreateTask.jsx:66]: for the handleChangeDueDate is is enough to use 
`setDueDate(selectedDate.toDate())`

Docs
https://day.js.org/docs/en/display/as-javascript-date

File: src/components/modal-content/CreateTask.jsx [hunk 6]
```diff
@@ -28,12 +28,57 @@ const OPTIONS = [
...
 	const theme = useTheme();
 
-	const handleChangeSelect = (event) => {
-		setSelect(event.target.value);
+	const handleChangeTaskName = (event) => {
+		setTaskName(event.target.value);
+	};
+	const handleChangePriority = (event) => {
+		setPriority(event.target.value);
+	};
```
Comment (cezarmocanu) [inline src/components/modal-content/CreateTask.jsx:65]: You can only have two handlers, one for the handleChangeDueDate and one for handleFieldChange as such

```js
const handleFieldChange = (event) => {
      setFormState({...formState, [event.target.name]: event.target.value});
};

const handleChangeDueDate = (selectedDate) => {
      setFormState({...formState, dueDate: selectedDate.toDate()});
};

```

Don't forget to add a `name` property to all textFields for this implementation to work

File: src/components/modal-content/CreateTask.jsx [hunk 7]
```diff
@@ -28,12 +28,57 @@ const OPTIONS = [
...
+	const handleChangeDescription = (event) => {
+		setDescription(event.target.value);
+	};
+
+	const TackDatails = {
+		taskName,
+		priority,
+		dueDate,
+		description,
+	};
```
Comment (cezarmocanu) [inline src/components/modal-content/CreateTask.jsx:78]: You can get rid of this as you will have the formData state

**Review by cezarmocanu — CHANGES_REQUESTED** [review]

File: src/components/modal-content/CreateTask.jsx [hunk 8]
```diff
@@ -3,6 +3,7 @@ import { useTheme } from "@mui/material/styles";
 import { useDispatch } from "react-redux";
```
Comment (cezarmocanu) [inline src/components/modal-content/CreateTask.jsx:3]: Also remove this import

**Review by cezarmocanu — CHANGES_REQUESTED** [review]

File: src/components/modal-content/CreateTask.jsx (hunk 8 above)
Comment (AleBarboni) [reply src/components/modal-content/CreateTask.jsx:3]: I use dispatch at:
onClick={() => dispatch(closeModal())}

Comment (AleBarboni) [reply src/components/modal-content/CreateTask.jsx:3]: without useDispatch the close button doesn't work

**Pull request merged**